TheThingsNetwork/lorawan-stack
# Remove `contact_info` from validation process

#### Summary



OS counterpart of https://github.com/TheThingsIndustries/lorawan-stack/pull/4012
#### Changes



- Add migration to create `email_validations` table
- Add `EmailValidationRegistry` to api
- Add `EmailValidationStore` interface to IS's store
- Update `webui` to call `EmailValidationRegistry` instead of `ContactInfoRegistry` when validating emails.

#### Testing



Unit tests and manual testing.

<details>
<summary> Test Steps </summary>

Use the following configuration for the Stack (remember to swap <field_ref> with an appropriate value):

```yml
 tls:
  source: file
  root-ca: ./ca.pem
  certificate: ./cert.pem
  key: ./key.pem
is:
  email:
    provider: "dir"
    dir: <dir_path> # (for example "./.env/emails/")
    sender-address: <sender_address>
    sender-name: <sender_name>
    templates:
      source: "dir"
      directory: "./pkg/email/templates"
  user-registration:
    email-validation:
      required: true
      retry-interval: 1h0m0s
      token-ttl: 48h0m0s
```

Follow the steps below:
1. Create two users.
2. Check the `<dir_path>` used in the config to validate if two emails were sent.
3. Select one of the emails and use the link to validate the user's account 
4. Select the other email and use the reference and token to validate the account via the CLI. 
5. Validate if both users have their emails validated.
    - Can be done via CLI with the command: `ttn-lw-cli usr get <usr_id> --primary-email-address --primary-email-address-validated-at`

</details>


##### Regressions



N/A

#### Checklist
